src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -190,10 +208,142 @@ public ResultSet getClientInfoProperties() throws SQLException {
         throw new SQLFeatureNotSupportedException("TODO");
     }
 
+    private MongoJsonSchema getFunctionJsonSchema() {
+        MongoJsonSchema schema = new MongoJsonSchema();
+        schema.bsonType = "object";
+        schema.additionalProperties = false;
+        schema.required = new HashSet<String>();
+        schema.properties = new HashMap<String, MongoJsonSchema>();
```

Comment:
Maybe on the `MongoJsonSchema` class we should have some additional constructor or some initialization method or some other static "constructor" method that directly returns to us this initialized, empty-object schema?

Something like
```
public MongoJsonSchema(String bsonType) {
    // if object: initialize like an object schema, else initialize with just bsonType set?
    ...
}

// or
public MongoJsonSchema(String bsonType, boolean additionalProperties) {
    ...
}

// alternatively
public static MongoJsonSchema makeEmptyObjectSchema {
    MongoJsonSchema s = new MongoJsonSchema();
    // initialize
    ...
}
```

We may be doing this type of initialization a lot.

Reply:
I like `makeEmptyObjectSchema`